### PR TITLE
Add some FS commands

### PR DESCRIPTION
**Pull request by Thealexbarney**
fsOpenCustomStorageFileSystem
fsExtendSaveDataFileSystem
fsSetGlobalAccessLogMode
fsGetGlobalAccessLogMode

### CODE REVIEW

File: nx/source/services/fs.c [hunk 1]
```diff
@@ -891,6 +932,123 @@ Result fsWriteSaveDataFileSystemExtraData(const void* buf, size_t len, FsSaveDat
...
+        resp = r.Raw;
+
+        rc = resp->result;
+    }
+
+    return rc;
+}
+
+Result fsSetGlobalAccessLogMode(u32 mode)
+{
```
Comment (yellows8) [inline nx/source/services/fs.c]: Move to previous line, same below.

File: nx/include/switch/services/fs.h [hunk 2]
```diff
@@ -272,6 +278,7 @@ Result fsOpenGameCardFileSystem(FsFileSystem* out, const FsGameCardHandle* handl
 Result fsReadSaveDataFileSystemExtraDataBySaveDataSpaceId(void* buf, size_t len, FsSaveDataSpaceId saveDataSpaceId, u64 saveID);
 Result fsReadSaveDataFileSystemExtraData(void* buf, size_t len, u64 saveID);
 Result fsWriteSaveDataFileSystemExtraData(const void* buf, size_t len, FsSaveDataSpaceId saveDataSpaceId, u64 saveID);
+Result fsExtendSaveDataFileSystem(FsSaveDataSpaceId saveDataSpaceId, u64 saveID, s64 dataSize, s64 journalSize);
```
Comment (yellows8) [inline nx/include/switch/services/fs.h]: Add comments for required sysver, same below.

**Comment by yellows8** [comment]
Can you doc what CustomStorage is on wiki?

**Comment by yellows8** [comment]
That wiki edit doesn't explain what "CustomStorage" actually is (as in more than just the enum).

**Comment by Thealexbarney** [comment]
I hadn't finished adding the rest yet. The other part is added now

File: nx/include/switch/services/fs.h [hunk 3]
```diff
@@ -272,6 +278,7 @@ Result fsOpenGameCardFileSystem(FsFileSystem* out, const FsGameCardHandle* handl
 Result fsReadSaveDataFileSystemExtraDataBySaveDataSpaceId(void* buf, size_t len, FsSaveDataSpaceId saveDataSpaceId, u64 saveID);
 Result fsReadSaveDataFileSystemExtraData(void* buf, size_t len, u64 saveID);
 Result fsWriteSaveDataFileSystemExtraData(const void* buf, size_t len, FsSaveDataSpaceId saveDataSpaceId, u64 saveID);
+Result fsExtendSaveDataFileSystem(FsSaveDataSpaceId saveDataSpaceId, u64 saveID, s64 dataSize, s64 journalSize); /// 3.0.0+
```
Comment (yellows8) [inline nx/include/switch/services/fs.h]: Is it known whether dataSize/journalSize used elsewhere are also signed?

Comment (Thealexbarney) [reply nx/include/switch/services/fs.h]: Yes. The fields in the save create struct also use signed division

Comment (yellows8) [reply nx/include/switch/services/fs.h]: Update those then?

Comment (Thealexbarney) [reply nx/include/switch/services/fs.h]: I didn't check FsSaveDataExtraData, but I'm assuming it's the same there

**Pull request merged**